server/src/graql/query/pattern/statement/StatementInstance.java

```diff
@@ -0,0 +1,371 @@
+/*
+ * GRAKN.AI - THE KNOWLEDGE GRAPH
+ * Copyright (C) 2018 Grakn Labs Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package grakn.core.graql.query.pattern.statement;
+
+import grakn.core.graql.query.Query;
+import grakn.core.graql.query.pattern.property.HasAttributeProperty;
+import grakn.core.graql.query.pattern.property.IdProperty;
+import grakn.core.graql.query.pattern.property.IsaExplicitProperty;
+import grakn.core.graql.query.pattern.property.IsaProperty;
+import grakn.core.graql.query.pattern.property.NeqProperty;
+import grakn.core.graql.query.pattern.property.RelationProperty;
+import grakn.core.graql.query.pattern.property.TypeProperty;
+import grakn.core.graql.query.pattern.property.ValueProperty;
+import grakn.core.graql.query.pattern.property.VarProperty;
+
+import javax.annotation.CheckReturnValue;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.joining;
+
+public abstract class StatementInstance extends Statement {
+
+    private static IllegalArgumentException illegalArgumentException(Statement statement, VarProperty varProperty) {
+        String message = "Not allowed to provide Statement Property: [" + varProperty.toString() + "] ";
+        message += "to " + statement.getClass().getSimpleName() + ": [" + statement.toString() + "]";
+        throw new IllegalArgumentException(message);
+    }
+
+    private StatementInstance(Variable var, LinkedHashSet<VarProperty> properties, Sign sign) {
+        super(var, properties, sign);
+    }
+
+    @SuppressWarnings("Duplicates")
```

Comment:
why supress?

Reply:
this was left over from when I was copying it from `Statement.java`.

Reply:
you can remove it